### ISSUE TITLE
bug in "get_metric_name" method.

### ISSUE THREAD

**Issue by evertonaleixo** [ContinualAI/avalanche#985]
The method "get_metric_name" of metric_utils, is returning the task ID in metric, even when we pass the parameter add_task=False.

When we pass False to this parameter, it considers False as an int instance and converts it to '0'. It enters in the last elif of the image below.

It makes the EarlyStoppingPlugin not work right, because of the metric's name.
Maybe, the EarlyStoppingPlugin should look to metrics using a pattern and not exactly metric name.

🐜 **To Reproduce**
```
## import torch.nn as nn

from torch.nn import CrossEntropyLoss
from torch.optim import SGD
from avalanche.evaluation.metrics import forgetting_metrics, accuracy_metrics, loss_metrics
from avalanche.evaluation.metrics import timing_metrics, cpu_usage_metrics
from avalanche.evaluation.metrics import confusion_matrix_metrics, disk_usage_metrics

from avalanche.benchmarks.classic import PermutedMNIST
from avalanche.benchmarks.scenarios import NCScenario
from avalanche.benchmarks.generators import benchmark_with_validation_stream

from avalanche.models import SimpleCNN, SimpleMLP
from avalanche.logging import TextLogger, TensorboardLogger, InteractiveLogger

from avalanche.training.plugins import EvaluationPlugin
from avalanche.training.strategies import Naive, JointTraining,BaseStrategy
from avalanche.training.plugins import EarlyStoppingPlugin

model = SimpleMLP(num_classes=10)

# Here we show all the MNIST variation we offer in the "classic" benchmarks
scenario = PermutedMNIST(n_experiences=2, seed=1)

# Than we can extract the parallel train and test streams
train_stream = scenario.train_stream
test_stream = scenario.test_stream

# Prepare for training & testing
optimizer = SGD(model.parameters(), lr=0.001, momentum=0.9)
criterion = CrossEntropyLoss()

# Continual learning strategy with default logger
esp = EarlyStoppingPlugin(3, 'test_stream')
cl_strategy = Naive(
    model, optimizer, criterion, train_mb_size=32, train_epochs=5,
    eval_mb_size=32, device='cuda', eval_every=1,
    plugins=[esp])

for train_task, test_task in zip(train_stream, test_stream):
    print("Current Classes: ", train_task.classes_in_this_experience)
    cl_strategy.train(train_task, eval_streams=[test_task])



from avalanche.evaluation.metric_utils import get_metric_name
x = accuracy_metrics(epoch=True, experience=True, stream=True)
print(get_metric_name(x[0], cl_strategy,add_experience=False, add_task=False))
```

🐝 **Expected behavior**
The metric name should be generated without task id.

🐞 **Screenshots**
![image](https://user-images.githubusercontent.com/6534458/163640126-10a78769-ce26-4170-8f9c-614f8da05b99.png)

![image](https://user-images.githubusercontent.com/6534458/163643857-f5546e59-4c8f-46de-a148-c6034d36fbd3.png)

**Comment by evertonaleixo** [comment]
I think this bug is related to issue #904 .

**Comment by AndreaCossu** [comment]
You are right, this function has the wrong behavior, since python considers boolean instances of both `bool` and `int`. I can fix the function, thanks for reporting!
